aws_lambda_powertools/metrics/metrics.py

```diff
@@ -69,76 +73,82 @@ def lambda_handler():
-
-    - Create metrics for distinct namespaces
-    - Create the same metrics with different dimensions more than once
-    """
+        if provider is None:
+            self.provider = AmazonCloudWatchEMFProvider(
+                namespace=namespace,
+                service=service,
+                metric_set=self.metric_set,
+                dimension_set=self.dimension_set,
+                metadata_set=self.metadata_set,
+                default_dimensions=self._default_dimensions,
+            )
+        else:
+            self.provider = provider
+
+    def add_metric(
+        self,
+        name: str,
+        unit: MetricUnit | str,
+        value: float,
+        resolution: MetricResolution | int = 60,
+    ) -> None:
+        self.provider.add_metric(name=name, unit=unit, value=value, resolution=resolution)
+
+    def add_dimension(self, name: str, value: str) -> None:
+        self.provider.add_dimension(name=name, value=value)
+
+    def serialize_metric_set(
+        self,
+        metrics: Dict | None = None,
+        dimensions: Dict | None = None,
+        metadata: Dict | None = None,
+    ) -> Dict:
+        return self.provider.serialize_metric_set(metrics=metrics, dimensions=dimensions, metadata=metadata)
+
+    def add_metadata(self, key: str, value: Any) -> None:
+        self.provider.add_metadata(key=key, value=value)
+
+    def flush_metrics(self, raise_on_empty_metrics: bool = False) -> None:
+        self.provider.flush_metrics(raise_on_empty_metrics=raise_on_empty_metrics)
+
+    def log_metrics(
+        self,
+        lambda_handler: Callable[[Dict, Any], Any] | Optional[Callable[[Dict, Any, Optional[Dict]], Any]] = None,
+        capture_cold_start_metric: bool = False,
+        raise_on_empty_metrics: bool = False,
+        default_dimensions: Dict[str, str] | None = None,
+    ):
+        return self.provider.log_metrics(
+            lambda_handler=lambda_handler,
+            capture_cold_start_metric=capture_cold_start_metric,
+            raise_on_empty_metrics=raise_on_empty_metrics,
+            default_dimensions=default_dimensions,
+        )
 
-    _dimensions: Dict[str, str] = {}
-    _default_dimensions: Dict[str, Any] = {}
+    def _extract_metric_resolution_value(self, resolution: Union[int, MetricResolution]) -> int:
+        return self.provider._extract_metric_resolution_value(resolution=resolution)
 
-    def __init__(self, service: Optional[str] = None, namespace: Optional[str] = None):
-        self.default_dimensions = self._default_dimensions
-        self.dimension_set = self._dimensions
+    def _extract_metric_unit_value(self, unit: Union[str, MetricUnit]) -> str:
+        return self.provider._extract_metric_unit_value(unit=unit)
 
-        self.dimension_set.update(**self._default_dimensions)
-        super().__init__(namespace=namespace, service=service)
+    def _add_cold_start_metric(self, context: Any) -> None:
+        self.provider._add_cold_start_metric(context=context)
```

Comment:
@heitorlessa We need to persist default dimension values in AmazonCloudWatchEMFProvider and Metrics.